sandcastle/api.py

```diff
@@ -678,17 +693,30 @@ def _copy_path_from_pod(self, local_dir: Path, pod_dir: Path):
         :param local_dir: path to the local dir
         :param pod_dir: path within the pod
         """
-        run_command(
-            [
-                "oc",
-                "rsync",
-                "--delete",  # delete files in local_dir which are not in pod_dir
-                "--quiet=true",  # avoid huge logs
-                f"--namespace={self.k8s_namespace_name}",
-                f"{self.pod_name}:{pod_dir}/",  # trailing / to copy only content of dir
-                f"{local_dir}",
-            ]
-        )
+        try:
+            run_command(
+                [
+                    "oc",
+                    "rsync",
+                    "--delete",  # delete files in local_dir which are not in pod_dir
+                    "--quiet=true",  # avoid huge logs
+                    f"--namespace={self.k8s_namespace_name}",
+                    f"{self.pod_name}:{pod_dir}/",  # trailing / to copy only content of dir
+                    f"{local_dir}",
+                ]
+            )
+        except Exception as ex:
+            # There is a race condition in k8s that it tells the pod is running even
+            # though it already killed an exec session and hence we couldn't copy
+            # anything from the pod
+            if not self.is_pod_running() or "not available in container" in str(ex):
+                logger.warning(
```

Comment:
Shouldn't this be an 'error', given that there is an exception raised in the next line?

Reply:
the only reason I used warning was that it wouldn't pop up in sentry when this happens for timed out runs